tests/unit/test_charm.py

```diff
@@ -264,56 +244,107 @@ def test_given_bind_address_not_available_when_evaluate_status_then_status_is_wa
             WaitingStatus("Waiting for bind and ingress addresses to be available"),
         )
 
+    @patch("charm.Vault", autospec=True)
     @patch("ops.model.Model.get_binding")
-    def test_given_not_leader_and_init_secret_not_set_when_evaluate_status_then_status_is_waiting(
-        self, patch_get_binding
+    def test_given_vault_uninitialized_when_evaluate_status_then_status_is_blocked(
+        self, mock_get_binding: MagicMock, mock_vault_class: MagicMock
```

Comment:
Not a critical thing, but we might want to use either the prefix mock or the prefix patch for consistency, instead of mixing both. This has been the case before too so I don't expect all names to get aligned in this PR :)

Reply:
the patch decorator replaces a real object with a different object, which in our case is always a mock object. So we have to either call everything a patch or everything a mock.

Reply:
I can mass replace all of them pretty easily with vim, but i think it makes sense to do it in a different PR since this one is very large.

Reply:
I agree, it's fine yo have it later